pipeline/meshing.cc

```diff
@@ -0,0 +1,138 @@
+// Author: Philipp Lindenberger (Phil26AT)
+#include "colmap/mvs/meshing.h"
+
+using namespace colmap;
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/stl_bind.h>
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+#include <string>
+
+#include "helpers.h"
+#include "log_exceptions.h"
+
+void init_meshing(py::module& m) {
+    using PoissonMOpts = mvs::PoissonMeshingOptions;
+    auto PyPoissonMeshingOptions =
+        py::class_<PoissonMOpts>(m, "PoissonMeshingOptions")
+            .def(py::init<>())
+            .def_readwrite("point_weight", &PoissonMOpts::point_weight,
+                "This floating point value specifies the importance that interpolation of"
+                "the point samples is given in the formulation of the screened Poisson"
+                "equation. The results of the original (unscreened) Poisson Reconstruction"
+                "can be obtained by setting this value to 0.")
+            .def_readwrite("depth", &PoissonMOpts::depth,
+                "This integer is the maximum depth of the tree that will be used for surface"
+                "reconstruction. Running at depth d corresponds to solving on a voxel grid"
+                "whose resolution is no larger than 2^d x 2^d x 2^d. Note that since the"
+                "reconstructor adapts the octree to the sampling density, the specified"
+                "reconstruction depth is only an upper bound.")
+            .def_readwrite("color", &PoissonMOpts::color,
+                "If specified, the reconstruction code assumes that the input is equipped"
+                "with colors and will extrapolate the color values to the vertices of the"
+                "reconstructed mesh. The floating point value specifies the relative"
+                "importance of finer color estimates over lower ones.")
+            .def_readwrite("trim", &PoissonMOpts::trim,
+                "This floating point values specifies the value for mesh trimming. The"
+                "subset of the mesh with signal value less than the trim value is discarded.")
+            .def_readwrite("num_threads", &PoissonMOpts::num_threads,
+                "The number of threads used for the Poisson reconstruction.");
```

Comment:
Is this formatting consistent with existing code in the repo? We should add a sensible `.clang-format` config.

Reply:
Yes this is more or less consistent with most parts, however some other code isn't and I agree that we should add a `.clang-format` config. We could use the [one from pixsfm](https://github.com/cvg/pixel-perfect-sfm/blob/main/.clang-format).

Reply:
Set up demo clang format config in #113